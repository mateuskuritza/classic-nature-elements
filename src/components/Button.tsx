import styled from "styled-components";

export default function Button(props: any) {
	const { children, ...rest } = props;
	return <ButtonStyle {...rest}>{children}</ButtonStyle>;
}

const ButtonStyle = styled.button`
	border: none;
	border-radius: 10px;
	padding: 5px;
	width: 80%;

	background-color: #397ca8;
	color: white;

	cursor: pointer;

	:hover {
		transform: scale(1.05);
		background-color: #599ecc;
	}
`;
