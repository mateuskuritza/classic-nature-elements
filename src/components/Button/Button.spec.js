import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Button from "./Button";

it('Should render the Button', () => {
    const { container } = render(<Button />)
    const button = screen.getByRole('button');
    expect(button).toBeDefined();
    expect(container).toMatchSnapshot();
});

it('Should render the Button with children', () => {
    const text = "testText";
    const { container } = render(<Button >{text}</Button>)
    const button = screen.getByText(text);
    expect(button.innerHTML).toBe(text);
    expect(container).toMatchSnapshot();
});


it('Should render the Button with props', () => {
    const text = "testText";
    const { container } = render(<Button placeholder={text}></Button>)
    const button = screen.getByPlaceholderText(text);
    expect(button.innerText).toBe(undefined);
    expect(container).toMatchSnapshot();
});

it('Should render the Button with children and props with function', () => {
    const text = "testText";
    let value = false
    const testFunction = () => {
        value = true;
    };

    const { container } = render(<Button placeholder={text} onClick={() => testFunction()}>{text}</Button>)

    const button = screen.getByPlaceholderText(text);
    expect(button.innerHTML).toBe(text);
    expect(container).toMatchSnapshot();

    fireEvent.click(button)
    expect(value).toBe(true);
});