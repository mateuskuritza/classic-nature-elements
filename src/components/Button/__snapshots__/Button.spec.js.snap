// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Button 1`] = `
<div>
  <button
    class="sc-bdvvtL LSrWP"
  />
</div>
`;

exports[`Should render the Button with children 1`] = `
<div>
  <button
    class="sc-bdvvtL LSrWP"
  >
    testText
  </button>
</div>
`;

exports[`Should render the Button with children and props with function 1`] = `
<div>
  <button
    class="sc-bdvvtL LSrWP"
    placeholder="testText"
  >
    testText
  </button>
</div>
`;

exports[`Should render the Button with props 1`] = `
<div>
  <button
    class="sc-bdvvtL LSrWP"
    placeholder="testText"
  />
</div>
`;
